engine/dlib/src/dlib/json.cpp

```diff
@@ -211,15 +207,24 @@ namespace dmJson
             return RESULT_OK;
         }
 
+        jsmn_parser parser;
         jsmnerr_t err = (jsmnerr_t) 0;
         jsmntok_t* tokens = 0;
-        do {
-            jsmn_init(&parser);
-            token_count += dmMath::Min(256U, token_count);
-            free(tokens);
+
+        jsmn_init(&parser);
+        int token_count = (uint32_t)jsmn_parse(&parser, buffer, buffer_length, 0, 0);
+        if (token_count >= 0)
```

Comment:
Not sure why I didn't see this last time I updated this code.
THe api _does_ have a way to query the number of required tokens.

4mb test file went from many minutes down to 1.4s.